python/rikai/tensorflow/pandas.py

```diff
@@ -0,0 +1,67 @@
+#  Copyright 2021 Rikai Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+# Standard
+from typing import Callable, Optional, Union
+
+# Third Party
+import pandas as pd
+import tensorflow as tf
+
+# Rikai
+from rikai.parquet.dataset import convert_tensor
+from rikai.spark.sql.codegen.base import unpickle_transform
+
+__all__ = ["PandasDataset"]
+
+from rikai.types import Image
+
+
+class PandasDataset:
+    """a Map-style Tensorflow dataset from a :py:class:`pandas.DataFrame` or a
+    :py:class:`pandas.Series`.
+
+    Note
+    ----
+
+    This class is used in Rikai's SQL-ML Spark implementation, which utilizes
+    pandas UDF to run inference.
+    """
+
+    def __init__(
+        self,
+        df: Union[pd.DataFrame, pd.Series],
+        transform: Optional[Callable] = None,
+        unpickle: bool = False,
+        use_pil: bool = False,
+    ) -> None:
+        assert isinstance(df, (pd.DataFrame, pd.Series))
+        self.df = df
+        self.transform = transform
+        self.unpickle = unpickle
+        self.use_pil = use_pil
+
+    def data(self, batch_size):
+
+        arr = self.df[0]
```

Comment:
From current tests, there is always only one item in `self.df`, not sure if it's the designed condition.

Reply:
we could double and double the dataset, and then, log the size of the pandas udf input

Reply:
you can use https://github.com/eto-ai/rikai/blob/v0.1.4/experimental/tfhub/tests/test_tfhub.py to run the test

Reply:
you can use https://github.com/eto-ai/rikai/blob/v0.1.4/experimental/tfhub/tests/test_tfhub.py to run the test

Reply:
this does not sound right. can we check batch size and etc. ?